### PR TITLE
Linkify administrative citations

### PR DESCRIPTION
**Pull request by v-anne**
This is addressing issue #4228, linking to administrative agency dockets.

Some cases in the federal court system are filed directly in the courts of appeals, bypassing the district courts. Often in those cases, a federal regulation or an administrative agency adjudication is being challenged. This change helps link to some administrative agency dockets, allowing for users to see the background of a case.

Right now, support for citations to rules published in the Federal Register is supported, as is the NLRB docket format. Other smaller agencies will be accounted for in a future update.

### CODE REVIEW

**Comment by mlissner** [comment]
Looks like MyPy is unhappy with your typing in the Linting Action. We'll want to fix that before we merge. Your branch is also a few commits behind `main`, but that's easier to fix.

I'll put this on Eduardo's backlog to review. 

Thank you for the hard work!

**Comment by v-anne** [comment]
@mlissner, I caught my mistake. All fixed. Should be good to merge!

**Comment by ERosendo** [comment]
LGTM 👍

**Review by ERosendo — APPROVED** [review]

File: cl/lib/model_helpers.py [hunk 1]
```diff
@@ -487,3 +487,36 @@ def suppress_autotime(model, fields):
...
                     "auto_now_add"
                 ]
+
+
+def linkify_orig_docket_number(agency: str, og_docket_number: str) -> str:
+    """Make an originating docket number for an appellate case into a link (MVP version)
+
+    :param agency: The administrative agency the case originated from
+    :param og_docket_number: The docket number where the case was originally heard.
+    :returns: A linkified version of the docket number for the user to click on, or the original if no link can be made.
```
Comment (mlissner) [inline cl/lib/model_helpers.py:497]:
```suggestion
    """Make an originating docket number for an appellate case into a link (MVP version)

    **NOTE: These links are presented to users and should be subject to strict security checks.**

    For example, each regex should be carefully written so it accepts only the narrowest of
    matches. The risk is that:

      - Mallory uploads a bad document via the RECAP APIs (these are open APIs).
      - The code here parses that upload in a way to create a redirect on the federalregister.gov
        website. 
      - federalregsiter.gov has an open redirect vulnerability (these are common).
      - The user clicks a link on our site that goes to federalregister.gov, which redirects the
        user to evilsite.com (b/c evilsite.com got through our checks here).
      - The user is tricked on that site into doing something bad.
  
    This is all quite unlikely, but we can ensure it doesn't happen by being strict about 
    the inputs our regular expressions capture.

    :param agency: The administrative agency the case originated from
    :param og_docket_number: The docket number where the case was originally heard.
    :returns: A linkified version of the docket number for the user to click on, or the original if no link can be made.
```

**Comment by mlissner** [comment]
Looks good. I added a big note about the danger of this code so that we can keep that in mind as we work on it. Looks fine for now, but want to be careful here, since it could cause trouble. 

Thank you @v-anne for the great contribution!

**Pull request merged**